src/packagedcode/opam.py

```diff
@@ -0,0 +1,249 @@
+
+
+def parse(location):
+    if not is_opam(location):
+        return
+
+    package_data = parse_opam(location)
+    return build_opam_package(package_data)
+
+
+def build_opam_package(package_data):
+    """
+    Return a Package object from a opam file or None.
+    """
+    package_dependencies = []
+    deps = package_data.get('depends') or []
+    for dep in deps:
+        package_dependencies.append(
+            models.DependentPackage(
+                purl=dep.purl,
+                requirement=dep.version,
+                scope='dependency',
+                is_runtime=True,
+                is_optional=False,
+                is_resolved=False,
+            )
+        )
+
+    name = package_data.get('name')
+    maintainers = package_data.get('maintainer') or []
+    authors = package_data.get('authors') or []
+    homepage_url = package_data.get('homepage')
+    vcs_url = package_data.get('dev-repo')
+    summary = package_data.get('synopsis')
+    description = package_data.get('description')
+    if summary:
+        description = summary
+    elif summary and description:
+        if len(summary) > len(description):
+            description = summary
+    
+    parties = []
+    for author in authors:
+        parties.append(
+            models.Party(
+                type=models.party_person,
+                name=author,
+                email=None,
+                role='author'
+            )
+        )
+    
+    for maintainer in maintainers:
+        parties.append(
+            models.Party(
+                type=models.party_person,
+                name=None,
+                email=maintainer,
+                role='email'
+            )
+        )
+
+    package = OpamPackage(
+        name=name or None,
+        vcs_url=vcs_url or None,
+        homepage_url=homepage_url,
+        description=description,
+        parties=parties,
+        dependencies=package_dependencies
+    )
+
+    return package
+
+
+@attr.s()
+class Opam(object):
+    name = attr.ib(default=None)
+    version = attr.ib(default=None)
+
+    @property
+    def purl(self):
+        return PackageURL(
+                    type='opam',
+                    name=self.name
+                ).to_string()
+
+
+# Regex expressions to parse file lines
+parse_file_line = re.compile(
+    r'(?P<key>^(.+?))'
+    r'\:\s*'
+    r'(?P<value>(.*))'
+).match
+
+parse_dep = re.compile(
+    r'^\s*\"'
+    r'(?P<name>[A-z0-9\-]*)'
+    r'\"\s*'
+    r'(?P<version>(.*))'
+).match
+
+
+def parse_opam(location):
+    """
+    Return a dictionary containing all the opam data.
+    """
+    with io.open(location, encoding='utf-8', closefd=True) as data:
+        lines = data.readlines()
+
+    opam_data = {}
+
+    for i, line in enumerate(lines):
+        parsed_line = parse_file_line(line)
+        if parsed_line:
+            key = parsed_line.group('key').strip()
+            value = parsed_line.group('value').strip()
+            if 'maintainer' == key:
```

Comment:
Please give examples on what this looks like in comments, so we can understand what and why the  code is this way

Reply:
:+1: